python/kserve/kserve/storage.py

```diff
@@ -181,7 +181,7 @@ def _download_s3(uri, temp_dir: str):
             target_key = (
                 obj.key.rsplit("/", 1)[-1]
                 if bucket_path == obj.key
-                else obj.key.replace(bucket_path, "", 1).lstrip("/")
+                else obj.key.replace(bucket_path.rsplit("/", 1)[0], "", 1).lstrip("/")
```

Comment:
@tjandy98 would this break the case `s3://test-bucket/a` that bucket path `test-bucket/a` becomes `test-bucket` after `rsplit` and object key `test-bucket/a/b/model.bin` is replaced with `dir/a/b/model.bin` which is supposed to be `dir/b/model.bin?

Reply:
>>> s = "test-bucket/a/b/model.bin"
>>> bucket = "test-bucket/a"
>>> s.replace(bucket_path.rsplit("/", 1)[0], "", 1).lstrip("/")
>>> bucket_path = "test-bucket/a"
>>> s.replace(bucket_path.rsplit("/", 1)[0], "", 1).lstrip("/")
'a/b/model.bin'
>>> s.replace(bucket_path, "", 1).lstrip("/")
'b/model.bin'

Reply:
@yuzisun You are right, thanks for providing an example! I have made some changes and added a test based on this.